### PR TITLE
TRC-52 - feat: add `TrendingSuggestions` component

### PR DESCRIPTION
**Pull request by mmcbride2929**
TL/DR
This story involves creating a top-level component TrendingSuggestions that fetches and handles data from MSW using Axios and React Query. 

Overview of Changes

`TrendingSuggestions.tsx`
- Use RQ to handle incoming `trending-suggestions` data 
- Add parameter `{ staleTime: Infinity }` to avoid re-fetching when users refocus the window
- Import & render `TrendingSuggestionList` component with it's appropriate props.

`TrendingSuggestions.api.ts`
- Created function to fetch mock data via an axios get request

`TrendingSuggestions.test.tsx`
- Wrote test to ensure our data is being rendered once the fetch completes.
- Add RQ / MSW functionality to the test

`mockServer.ts`
- add MSW server for global imports

`apps/web/package.json`
- Add moduleNameMapper to enable jest to interact with ES6 imports.

`TrendingSuggestionList.tsx`
- Add export to interface to use it within `TrendingSuggestions.tsx`
- Refactor subRedditIcon to string within `metaDataItem` interface

`TrendingSuggestionList.utils.tsx`
- Remove Icon & replace typing with string

`TrendingItemSuggestion.tsx`
- Refactor interface to accept temporary string value for subRedditIcon 
- Remove Icon component (temporary)

`testSetup.tsx`
- Add global React Query provider for test components

### CODE REVIEW

File: apps/web/src/components/TrendingItemSuggestion/TrendingItemSuggestion.tsx [hunk 1]
```diff
@@ -10,7 +10,7 @@ interface Props {
     trendingSuggestionHeading: string;
     trendingSuggestionSummary: string;
     targetSubreddit: {
-      subRedditIcon: React.ReactElement;
+      subRedditIcon: string;
```
Comment (willcary) [inline apps/web/src/components/TrendingItemSuggestion/TrendingItemSuggestion.tsx:13]: Just a reminder for us all to make a ticket to figure out how to unstringify the icons after being fetched with axios. Then we could add these icons back in.

File: apps/web/src/components/TrendingSuggestionList/TrendingSuggestionList.utils.tsx [hunk 2]
```diff
@@ -1,17 +1,10 @@
-import Icon from '../Icon/Icon';
-import { ReactComponent as ReactIcon } from '../../assets/reactSubRedditIcon.svg';
-
 export const metaDataItems = [
```
Comment (willcary) [inline apps/web/src/components/TrendingSuggestionList/TrendingSuggestionList.utils.tsx:1]: You could consolidate these items into a `metaDataItem` since they are identical and then use that variable 3 times in the `metaDataItems` array. Then you would only need to make one change. Since this is just mock data though, I don't have strong feelings about doing this, just thought I would point it out.

File: apps/web/src/components/TrendingSuggestions/TrendingSuggestions.api.ts [hunk 3]
```diff
@@ -0,0 +1,15 @@
+import * as React from 'react';
+import axios from 'axios';
```
Comment (willcary) [inline apps/web/src/components/TrendingSuggestions/TrendingSuggestions.api.ts:2]: I've never used axios before, seems like a great tool, nice job!

File: apps/web/src/components/TrendingSuggestions/TrendingSuggestions.tsx [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+  );
+
+  // placeholder func, will be removed when we add react router + refactor
+  const handleClick = () => {
+    console.log('temp');
+  };
+
+  return (
+    <>
+      {data && (
```
Comment (willcary) [inline apps/web/src/components/TrendingSuggestions/TrendingSuggestions.tsx:23]: Won't this disable the 'loading' UI of your `TrendingSuggestionList` since the component will only render if the data is truthy, hence already loaded?

File: apps/web/src/components/TrendingSuggestions/TrendingSuggestions.test.tsx [hunk 5]
```diff
@@ -0,0 +1,80 @@
...
+import { QueryClient, QueryClientProvider } from 'react-query';
+
+const queryClient = new QueryClient();
+
+const server = setupServer(
+  rest.get('trending-suggestions', (req, res, ctx) => {
+    return res(
+      ctx.json([
+        {
+          URL: 'https://images.unsplash.com/photo-1515879218367-8466d910aaa4?ixlib=rb-4.0.3&ixid=MnwxMjA3fDB8MHxzZWFyY2h8MTB8fGNvZGV8ZW58MHx8MHx8&w=1000&q=80',
```
Comment (willcary) [inline apps/web/src/components/TrendingSuggestions/TrendingSuggestions.test.tsx:14]: Same comment as above about considering reusing one previously defined variable within this array since they are all the same. Actually in this case it would be better to use a function that accepts the heading number argument.

File: apps/web/src/components/TrendingSuggestions/TrendingSuggestions.test.tsx [hunk 6]
```diff
@@ -0,0 +1,80 @@
+import { rest } from 'msw';
+import { setupServer } from 'msw/node';
+import { render, waitFor, screen } from '@testing-library/react';
+import TrendingSuggestions from './TrendingSuggestions';
+import { QueryClient, QueryClientProvider } from 'react-query';
+
+const queryClient = new QueryClient();
+
+const server = setupServer(
```
Comment (willcary) [inline apps/web/src/components/TrendingSuggestions/TrendingSuggestions.test.tsx:9]: using msw and react query is all new to me, but I'm wondering if it might be better to set up the server somehwere more global so that it is accesible for a Storybook story or other use cases. Regardless, awesome job using all these technologies, I'm impressed!

**Review by willcary — COMMENTED** [review]
Great job using so many different technologies, Michael! One general question, should there be a Storybook file in this too, or is that omitted because this story doesn't actually add any new components to be shown in the UI?

File: apps/web/src/components/TrendingSuggestionList/TrendingSuggestionList.utils.tsx (hunk 2 above)
Comment (mmcbride2929) [reply apps/web/src/components/TrendingSuggestionList/TrendingSuggestionList.utils.tsx:1]: That's a good point. I thought about doing that as well, but kept getting an error when testing the component. The error was caused by having multiple elements with the same output in the heading property, but I recently learned I could have used the "allBy" query to fix that.

File: apps/web/src/mocks/mockServer.ts [hunk 7]
```diff
@@ -0,0 +1,11 @@
+import { rest } from 'msw';
+import { setupServer } from 'msw/lib/node';
+import { metaData } from '../components/TrendingItemSuggestion/TrendingItemSuggestion.utils';
+
+const server = setupServer(
```
Comment (willcary) [inline apps/web/src/mocks/mockServer.ts:5]: I spun up the `TrendingSuggestions` component in the app and it's working really well, fantastic! I noticed there is some possible overlap between this file and the `handlers.ts` file. Do they both do different things? I'm just not sure how they both work, so thought I'd ask.

**Review by willcary — COMMENTED** [review]
LGTM, just left one question out of curiosity. Nice work!

File: apps/web/src/components/TrendingItemSuggestion/TrendingItemSuggestion.tsx [hunk 8]
```diff
@@ -53,14 +53,13 @@ const TrendingItemSuggestion = ({
...
-              {trendingSuggestionSummary}
-            </h2>
+            <h2 data-testid="list-item-summary">{trendingSuggestionSummary}</h2>
             <div className="list-item-target-subreddit">
-              <Icon customTestId="list-item-subreddit-icon">
+              {/* <Icon customTestId="list-item-subreddit-icon">
+                
                 {subRedditIcon}
-              </Icon>
+              </Icon> */}
```
Comment (wijohnst) [inline apps/web/src/components/TrendingItemSuggestion/TrendingItemSuggestion.tsx:61]: Can we remove this commented code?

File: apps/web/src/components/TrendingSuggestions/TrendingSuggestions.api.ts [hunk 9]
```diff
@@ -0,0 +1,15 @@
...
+
+const fetchTrendingSuggestions = async () => {
+  try {
+    const response = await axios.get('/trending-suggestions');
+    console.log(response.data);
+    return response.data;
+  } catch (error) {
+    console.log(error);
```
Comment (wijohnst) [inline apps/web/src/components/TrendingSuggestions/TrendingSuggestions.api.ts:10]: Can we remove these log statements?

File: apps/web/src/components/TrendingSuggestions/TrendingSuggestions.test.tsx [hunk 10]
```diff
@@ -0,0 +1,30 @@
...
+import { QueryClient, QueryClientProvider } from 'react-query';
+import server from '../../mocks/mockServer';
+
+const queryClient = new QueryClient();
+
+const QueriedComponent = () => {
+  return (
+    <QueryClientProvider client={queryClient}>
+      <TrendingSuggestions />
+    </QueryClientProvider>
```
Comment (wijohnst) [inline apps/web/src/components/TrendingSuggestions/TrendingSuggestions.test.tsx:12]: Is there a way to abstract this out to our global `render` method?

https://testing-library.com/docs/react-testing-library/setup/

File: apps/web/src/components/TrendingSuggestions/TrendingSuggestions.tsx [hunk 11]
```diff
@@ -0,0 +1,32 @@
...
+import fetchTrendingSuggestions from './TrendingSuggestions.api';
+import TrendingSuggestionList from '../TrendingSuggestionList/TrendingSuggestionList';
+
+const TrendingSuggestions = () => {
+  const { data, isLoading } = useQuery(
+    ['trending-suggestions'],
+    () => {
+      const response = fetchTrendingSuggestions();
+      return response;
+    },
```
Comment (wijohnst) [inline apps/web/src/components/TrendingSuggestions/TrendingSuggestions.tsx:12]:
```suggestion
    () => fetchTrendingSuggestions()
```

File: apps/web/src/components/TrendingSuggestions/TrendingSuggestions.tsx [hunk 12]
```diff
@@ -0,0 +1,32 @@
...
+    { staleTime: Infinity }
+  );
+
+  // placeholder func, will be removed when we add react router + refactor
```
Comment (wijohnst) [inline apps/web/src/components/TrendingSuggestions/TrendingSuggestions.tsx:16]: In a comment like this that references tech debt, it's always a good idea to link to the ticket that fixes the issue.

**Review by wijohnst — COMMENTED** [review]
Left a few comments.

File: apps/web/src/components/TrendingSuggestions/TrendingSuggestions.test.tsx (hunk 10 above)
Comment (mmcbride2929) [reply apps/web/src/components/TrendingSuggestions/TrendingSuggestions.test.tsx:12]: Good call

File: apps/web/src/mocks/mockServer.ts (hunk 7 above)
Comment (mmcbride2929) [reply apps/web/src/mocks/mockServer.ts:5]: After going back and looking at `mockServer` & `browser` I thought the same thing. I tried re-configuring the `TrendingSuggestions.test.tsx` file so that it would depend on the browser request, but couldn't get it to work. After a little research it looks like you need to create a separate server for your test cases, and I created the separate file so we didn't have to define a new server in every component that depends on a MSW response.

**Review by willcary — COMMENTED** [review]
Great job, Michael! Left one comment about removing the `TrendingSuggestions` from the main `App`, but otherwise LGTM.

File: apps/web/src/components/TrendingSuggestions/TrendingSuggestions.tsx [hunk 13]
```diff
@@ -0,0 +1,30 @@
...
+  // ticket: https://github.com/users/wijohnst/projects/10/views/1?pane=issue&itemId=27392872
+  const handleClick = () => {
+    console.log('temp');
+  };
+
+  return (
+    <>
```
Comment (wijohnst) [inline apps/web/src/components/TrendingSuggestions/TrendingSuggestions.tsx:20]: NIT: Does this need to be wrapped in `<>`?

File: apps/web/src/test/testSetup.tsx
Comment (wijohnst) [inline apps/web/src/test/testSetup.tsx:1]: Great work on the `testSetup` stuff. This is a super important skill and something that most junior devs won't have touched. Keep this in mind at interview time. Nice work!

**Review by wijohnst — APPROVED** [review]
This looks great!
